sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -353,6 +354,16 @@ def create_entity(
             return _trim_service_metadata(metadata)
         except ResourceNotFoundError as error:
             _process_table_error(error)
+        except HttpResponseError as error:
```

Comment:
Actually it looks like there might have been a bug here... line 355 should not be catching `ResourceNotFoundError`, _all_ HttpResponseErrors need to go through decoding in `_process_table_error` - as you should not need to do that awkward code below in order to get the error code.